### PR TITLE
a zillion flakes

### PR DESCRIPTION
**Pull request by jbrockmendel**
ugh.  No doubt I missed a bunch...

There are a handful of things, especially in parsers.pyx where "comments" look like commented-out code that should be either deleted or otherwise handled.  I'm declaring that [SEP](http://hitchhikers.wikia.com/wiki/Somebody_Else%27s_Problem_field).

### CODE REVIEW

File: pandas/_libs/interval.pyx [hunk 1]
```diff
@@ -40,9 +41,9 @@ cdef class IntervalMixin:
 
 
 cdef _interval_like(other):
-    return (hasattr(other, 'left')
-            and hasattr(other, 'right')
-            and hasattr(other, 'closed'))
+    return (hasattr(other, 'left') and
+            hasattr(other, 'right') and
+            hasattr(other, 'closed'))
```
Comment (jorisvandenbossche) [inline pandas/_libs/interval.pyx]: Can we keep the original code and add this to the ignored ones (it's error code W503)? 
I much prefer having the 'and' on the new line, and the most recent PEP8 agrees with me (https://www.python.org/dev/peps/pep-0008/#should-a-line-break-before-or-after-a-binary-operator)

File: pandas/_libs/tslibs/frequencies.pyx [hunk 2]
```diff
@@ -10,89 +10,15 @@ np.import_array()
...
-    freqstr : str or tuple
-
-    Returns
-    -------
-    return : tuple of base frequency code and stride (mult)
-
-    Example
-    -------
-    >>> get_freq_code('3D')
-    (6000, 3)
```
Comment (jreback) [inline pandas/_libs/tslibs/frequencies.pyx]: duplicated code yes?

File: pandas/_libs/tslibs/frequencies.pyx [hunk 3]
```diff
@@ -10,89 +10,15 @@ np.import_array()
...
-    --------
-    _freq_and_stride('5Min') -> 'Min', 5
-    """
-    groups = opattern.match(freqstr)
-
-    if not groups:
-        raise ValueError("Could not evaluate {freq}".format(freq=freqstr))
-
-    stride = groups.group(1)
-
```
Comment (jreback) [inline pandas/_libs/tslibs/frequencies.pyx]: same?

**Comment by jreback** [comment]
fine modulo @jorisvandenbossche comment.

**Comment by jreback** [comment]
> ugh. No doubt I missed a bunch...

its pretty easy to run this locally.

File: pandas/_libs/interval.pyx (hunk 1 above)
Comment (jbrockmendel) [reply pandas/_libs/interval.pyx]: Good call!

**Comment by jbrockmendel** [comment]
We may need to re-think this strategy if we can't close this quickly.  This touches so many things that it's going to need to be rebased a ton.

**Comment by jbrockmendel** [comment]
Fixed up remaining complaints locally, with the exception of Timedelta.  Didn't want to cause merge conflicts with #17937.

**Comment by jreback** [comment]
@jbrockmendel 

The simplest method is to add 2 sections in lint.sh, one with the 'fixed' ones and one with the todo.

you leave the todo with the existing flake checks, and gradually add files to the fixed.

The you can change 1 file at a time (or a small number).

we did this for the entire code base; if you do it piece by piece its easy.

File: pandas/_libs/interval.pyx (hunk 1 above)
Comment (jorisvandenbossche) [reply pandas/_libs/interval.pyx]: Thanks for adding it to the ignore list. 
Can you undo the changes above? (and there are some below as well)

**Comment by jorisvandenbossche** [comment]
In principle this can be merged quickly, as you prefer (I think it are mainly your PRs that will have merge conflicts)

File: pandas/_libs/interval.pyx (hunk 1 above)
Comment (jbrockmendel) [reply pandas/_libs/interval.pyx]: Will do.

File: ci/lint.sh [hunk 4]
```diff
@@ -10,21 +10,21 @@ if [ "$LINT" ]; then
...
-    flake8 pandas --filename=*.py --exclude pandas/_libs/src
+    flake8 pandas --filename=*.py --exclude pandas/_libs/src --ignore=W503,E731
     if [ $? -ne "0" ]; then
         RET=1
     fi
     echo "Linting *.py DONE"
 
     echo "Linting setup.py"
-    flake8 setup.py
+    flake8 setup.py --ignore=W503
```
Comment (jreback) [inline ci/lint.sh]: you can just add the global ignores to ``setup.cfg``

File: pandas/_libs/tslibs/parsing.pyx [hunk 5]
```diff
@@ -44,10 +44,14 @@ from dateutil.relativedelta import relativedelta
...
 
+# ----------------------------------------------------------------------
+# Constants
+
 
 class DateParseError(ValueError):
     pass
 
+
 _nat_strings = set(['NaT', 'nat', 'NAT', 'nan', 'NaN', 'NAN'])
```
Comment (jreback) [inline pandas/_libs/tslibs/parsing.pyx]: add to list to incorporate from nattype

File: pandas/tests/io/test_feather.py [hunk 6]
```diff
@@ -1,17 +1,17 @@
 """ test feather-format compat """
-
```
Comment (jreback) [inline pandas/tests/io/test_feather.py]: was this not being linted before?

**Review by jreback — APPROVED** [review]
ok, pls don't add anything else.

File: pandas/tests/io/test_feather.py (hunk 6 above)
Comment (jbrockmendel) [reply pandas/tests/io/test_feather.py]: Just looked at config.cfg.  The flake complaint being fixed here is E402 (module-level import not at top of module), which is one of two specifically ignored in setup.cfg.  Woops. Just moved the ignore spec from lint.sh to cfg, will stop messing with this until further notice.

**Comment by jreback** [comment]
ok since we merged the nattype, this needs a rebase (upside is it should be good to go after) :>

ping when green.

**Comment by jreback** [comment]
thanks!

**Pull request merged**